contracts/src/interfaces/IStakeTable.sol

```diff
@@ -0,0 +1,127 @@
+    /// @param exitEpoch The ending epoch for the validator.
+    struct Node {
+        address account;
+        StakeType stakeType;
+        uint64 balance;
+        uint64 registerEpoch;
+        uint64 exitEpoch;
+    }
+
+    // === Table State & Stats ===
+
+    /// @notice Get the commitment of the stake table used in current voting (i.e. snapshot at
+    /// the start of last epoch)
+    function votingStakeTableCommitment() external view returns (bytes32);
+    /// @notice Get the commitment of the stake table frozen for change (i.g. snapshot at the start
+    /// of the current epoch)
+    function frozenStakeTableCommitment() external view returns (bytes32);
+
+    /// @notice Get the total stake of the registered keys in the latest stake table (Head)
+    function totalStake() external view returns (uint256);
+    /// @notice Get the total number of stakers, uniquely identified by their `blsVK`
+    function totalKeys() external view returns (uint32);
+    /// @notice Get the total stake of the registered keys in the voting stake table
+    /// (LastEpochStart)
+    function totalVotingStake() external view returns (uint256);
+
+    /// @notice Look up the balance of `blsVK`
+    function lookup(BN254.G1Point calldata blsVK) external view returns (uint64);
+    /// @notice Look up the full `Node` state associated with `blsVK`
+    function fullLookup(BN254.G1Point calldata blsVK) external view returns (Node memory);
+
+    // === Queuing Stats ===
+
+    /// @notice Get the next available epoch for new registration
+    function nextRegisterationEpoch() external view returns (uint64);
+    /// @notice Get the number of pending registration requests in the waiting queue
+    function numPendingRegistrations() external view returns (uint64);
+    /// @notice Get the next available epoch for exit
+    function nextExitEpoch() external view returns (uint64);
+    /// @notice Get the number of pending exit requests in the waiting queue
+    function numPendingExit() external view returns (uint64);
+
+    // === Write APIs ===
+
+    /// @notice Register a validator in the stake table, transfer of tokens incurred!
+    ///
+    /// @param blsVK The BLS verification key
+    /// @param schnorrVK The Schnorr verification key (as the auxiliary info)
+    /// @param amount The amount to register
+    /// @param stakeType The type of staking (native or restaking)
+    /// @param blsSig The BLS signature that authenticates the `blsVK` field
+    /// @param maxWait The maximum epoch the sender is waiting to wait to be included (cannot be
+    /// smaller than the current epoch)
+    ///
+    /// @return success status
+    ///
+    /// @dev No validity check on `schnorrVK`, as it's assumed to be sender's responsibility,
+    /// the contract only treat it as auxiliary info submitted by `blsVK`.
+    function register(
+        BN254.G1Point calldata blsVK,
+        EdOnBN254.EdOnBN254Point calldata schnorrVK,
+        uint64 amount,
```

Comment:
Is `uint64` type sufficient for `amount`?

Reply:
I'm following our [doc here](https://docs.espressosys.com/sequencer/~/revisions/oijyFXG7VlZ4pxqo4Lgq/espresso-sequencer-architecture/internal-functionality/sequencer-contract/stake-table#data-structures) by @jbearer 

64 bits should be enough?